src/main/java/org/apache/sling/resourceresolver/impl/mapping/MapEntries.java

```diff
@@ -714,6 +783,17 @@ public void onChange(final List<ResourceChange> changes) {
                 continue;
             }
 
+            // during startup: just enqueue the events
```

Comment:
It's maybe better to do the check at method start and then handle all changes in the same way. If the check is done here, the first changes might go into the queue, while the remaining ones are not.

Reply:
that could only happen when "vanityPathsProcessed" ever changes from true to false, right?

Reply:
right